trillian/ctdns/handlers.go

```diff
@@ -0,0 +1,311 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These handlers implement the CT over DNS proposal documented at:
+// https://github.com/google/certificate-transparency-rfcs/blob/master/dns/draft-ct-over-dns.md
+
+package ctdns
+
+import (
+	"context"
+	"encoding/base32"
+	"encoding/base64"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/google/certificate-transparency-go/trillian/ctfe"
+	"github.com/google/certificate-transparency-go/trillian/ctfe/configpb"
+	"github.com/google/trillian"
+	"github.com/google/trillian/types"
+	"github.com/miekg/dns"
+)
+
+// The maximum length of the text returned in a DNS consistency or inclusion
+// proof response.
+const maxProofLen = 255
+
+var (
+	// In these formats the last (.*) is always the zone, this is checked outside
+	// the regex matching.
+	sthRE     = regexp.MustCompile("^sth\\.(.*)\\.$")
```

Comment:
Back quotes would probably be easier to read.

Reply:
Done.